backend/src/main/java/com/huellapositiva/application/controller/NewsletterController.java

```diff
@@ -0,0 +1,79 @@
+package com.huellapositiva.application.controller;
+
+import com.huellapositiva.domain.actions.ChangeStatusNewsletterSubscriptionAction;
+import com.huellapositiva.domain.actions.ManageNewsletterExcelAction;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.*;
+
+import javax.annotation.security.RolesAllowed;
+import java.io.IOException;
+
+@RestController
+@AllArgsConstructor
+@Tag(name = "Newsletter ", description = "The newsletter manager API")
+@RequestMapping("/api/v1/newsletter")
+public class NewsletterController {
+
+    @Autowired
+    private final ManageNewsletterExcelAction manageNewsletterExcelAction;
+
+    @Autowired
+    private final ChangeStatusNewsletterSubscriptionAction changeStatusNewsletterSubscriptionAction;
+
+    @Operation(
+            summary = "Change status of subscription to newsletter",
+            description = "Changes the status of the subscribed parameter on the specified volunteer",
+            tags = "newsletter"
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(
+                            responseCode = "200",
+                            description = "Ok, status of subscribed field changed successfully"
+                    ),
+                    @ApiResponse(
+                            responseCode = "500",
+                            description = "Internal server error, could not fetch the user data due to a connectivity issue."
+                    )
+            }
+    )
```

Comment:
Check it with the new swagger. (VolunteerApiController)

Reply:
Solved